File: Design Patterns/Behavioral/Strategy/Case 1/Python/Duck.py
from Quak.QuackBehavior import QuackBehavior
from Fly.FlyBehavior import FlyBehavior
from DuckInterface import DuckInterface
class Duck(DuckInterface):
    
    def __init__(self) -> None:
        self.__flyBehavior:FlyBehavior=None
        self.__quakBehavior:QuackBehavior= None
    

    def setQuakBehavior(self,quak:QuackBehavior):
        self.__quakBehavior=quak


    def setFlyBehavior(self,fly:FlyBehavior):
        self.__flyBehavior=fly
    

    def getFlyBehavior(self)->FlyBehavior:
        return self.__flyBehavior
    

    def getQuakBehavior(self)->QuackBehavior:
        return self.__quakBehavior


    def dispay():
        pass

    def swim():
        pass

File: Design Patterns/Behavioral/Strategy/Case 1/PHP/Duck.php
<?php
namespace PHP;
// include_once("require.php");
require_once("Quak/QuakBehavior.php");
use PHP\Quak\QuakBehavior;
require_once("Fly/FlyBehavior.php");
use PHP\Fly\FlyBehavior;

abstract class Duck{
    private QuakBehavior $quakBehavior;
    private FlyBehavior $flyBehavior;

    function setFlyBehavior(FlyBehavior $flyBehavior){
        $this->flyBehavior=$flyBehavior;
    }

    function setQuakBehavior(QuakBehavior $quakBehavior){
        $this->quakBehavior=$quakBehavior;
    }

    function getQuakBehavior():QuakBehavior{
        return $this->quakBehavior;
    }

    function getFlyBehavior():FlyBehavior{
        return $this->flyBehavior;
    }


    function swim(){
        echo "Swimming";
    }

    abstract function display();
}